src/main/java/facade/MoveFacade.java

```diff
@@ -1,9 +1,120 @@
+
+    /**
+     * default constructor
+     */
+    public MoveFacade(){
+        clientConnectionManager = ClientConnectionManager.getInstance();
+        moveDao = MoveDao.getInstance();
+        cookieManager = CookieManager.getInstance();
+        gameDao = GameDao.getInstance();
+    }
+
+    /**
+     * constructor for unit tests
+     */
+    public MoveFacade(ClientConnectionManager clientConnectionManager, MoveDao moveDao, CookieManager cookieManager, GameDao gameDao){
+        this.clientConnectionManager = clientConnectionManager;
+        this.moveDao = moveDao;
+        this.cookieManager = cookieManager;
+        this.gameDao = gameDao;
+    }
+
+    /**
+     * Gets all moves for a specified game.
+     *
+     * @param gameId non-negative integer corresponding to the game.
+     * @param userCookie non-null String
+     * @param gameCookie non-null String
+     * @return list of all moves for the requested game.
+     * @throws GameNotFoundException when the game does not exist.
+     * @throws UserNotFoundException if cookie is for a different user.
+     * @throws InvalidUserCookieException if not a valid user cookie
+     * @throws InvalidGameCookieException if not a valid game cookie
+     */
+    public List<Move> getMovesForGame(int gameId, String userCookie, String gameCookie) throws InvalidGameCookieException, GameNotFoundException, InvalidUserCookieException, UserNotFoundException {
+        LOGGER.info("MoveFacade.getMovesForGame");
+        cookieManager.validateCookies(userCookie, gameCookie);
+        if(!cookieManager.getQualifier(gameCookie).equals(gameId + "")){
+            throw new InvalidGameCookieException("Game cookie doesn't match game id you are requesting");
+        }
+        return moveDao.getMovesForGame(gameId);
+    }
+
+    //make move
+    /**
+     * sends the move to the other player to be verified
+     * @param move non null Move attempting to be made.
+     * @param userCookie non-null String
+     * @param gameCookie non-null String
+     * @throws GameNotFoundException when the game does not exist.
+     * @throws UserNotFoundException if cookie is for a different user.
+     * @throws InvalidUserCookieException if not a valid user cookie
+     * @throws InvalidGameCookieException if not a valid game cookie
+     */
+    public void makeMove(Move move, String userCookie, String gameCookie) throws InvalidGameCookieException, GameNotFoundException, InvalidUserCookieException, UserNotFoundException {
+        cookieManager.validateCookies(userCookie, gameCookie);
+        String requestingUser = cookieManager.getQualifier(userCookie);
+        Game game = gameDao.getGame(Integer.parseInt(cookieManager.getQualifier(gameCookie)));
+        String otherPlayer;
+        if(game.getPlayer1().equals(requestingUser)){
+            otherPlayer = game.getPlayer2();
+        }else {
+            otherPlayer = game.getPlayer1();
+        }
+        movesToVerify.put(otherPlayer, move);
```

Comment:
Should we have the key for `movesToVerify` be the `gameId` instead of the username? If a player is a part of multiple games we could run into problems (theoretically).